pkg/graveler/graveler.go

```diff
@@ -509,6 +511,13 @@ type VersionController interface {
 	// Note: Ancestors of previously expired commits may still be considered if they can be reached from a non-expired commit.
 	SaveGarbageCollectionCommits(ctx context.Context, repository *RepositoryRecord, previousRunID string) (garbageCollectionRunMetadata *GarbageCollectionRunMetadata, err error)
 
+	// GetGarbageCollectionUncommitted Returns a list of at most GCMaxUncommittedObjects uncommitted objects, and a continuation token in the form of GCUncommittedMark if needed.
+	GetGarbageCollectionUncommitted(ctx context.Context, repository *RepositoryRecord, mark GCUncommittedMark) ([]*ValueRecord, *GCUncommittedMark, error)
+
+	// SaveGarbageCollectionUncommitted Uploads parquet file of uncommitted objects to the repository in the run ID path. Return the GC run's metadata
+	// including the run ID and the location of the saved parquet file
+	SaveGarbageCollectionUncommitted(ctx context.Context, repository *RepositoryRecord, filename, runID string) (*GarbageCollectionRunMetadata, error)
```

Comment:
IIUC it uploads to the file to the object store, right? Why is it a graveler functionality? We have the storage namespace in the repo record, we can use that and uplod directly to the store.

Reply:
This is a strange method signature for graveler: it basically gets a file and saves it in S3. It doesn't really do anything GC-related.

Reply:
@N-o-Z think you missed this

Reply:
I prefer the garbage collection manager handling the uploading of files to its managed prefix. Catalog seems like a too far high level place to decide where to write GC information. I can expose GetUncommittedLocation through graveler but it feels like a stretch. You could argue the same thing about SaveGarbageCollectionCommits (basically all the logic can be done in graveler)